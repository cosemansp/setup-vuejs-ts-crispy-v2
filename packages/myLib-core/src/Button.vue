<template>
  <button class="button" :style="{color: color, borderColor: color}" @click="onClick" :class="{rounded: rounded}">
    <slot></slot>!</button>
</template>

<script lang="ts">
import Vue from 'vue';
export default Vue.extend({
  props: {
    rounded: Boolean,
    color: {
      default: '#42b983',
    },
  },
  methods: {
    onClick(event) {
      this.$emit('clicked', event);
    },
  },
});
</script>

<style>
.rounded {
  border-radius: 5px;
}
.button {
  border: 3px solid;
  padding: 10px 20px;
  background-color: white;
  outline: none;
}
</style>
