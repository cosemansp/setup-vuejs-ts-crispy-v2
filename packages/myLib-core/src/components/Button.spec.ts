import { shallow } from '@vue/test-utils';
import Button from './Button.vue';

describe('Button.vue', () => {
  test('default props', () => {
    const button = new Button();
    expect(button.rounded).toBe(false);
    expect(button.color).toBe('#42b983');
  });

  test('default render', () => {
    const wrapper = shallow(Button);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders props.rounded when passed', () => {
    const wrapper = shallow(Button, { propsData: { rounded: true } });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders props.color when passed', () => {
    const wrapper = shallow(Button, { propsData: { color: '#ff0000' } });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
