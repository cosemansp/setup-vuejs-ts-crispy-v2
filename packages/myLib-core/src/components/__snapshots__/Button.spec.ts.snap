// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.vue default render 1`] = `<button class="button" style="color: rgb(66, 185, 131); border-color: #42b983;"></button>`;

exports[`Button.vue renders props.color when passed 1`] = `<button class="button" style="color: rgb(255, 0, 0); border-color: #ff0000;"></button>`;

exports[`Button.vue renders props.rounded when passed 1`] = `<button class="button rounded" style="color: rgb(66, 185, 131); border-color: #42b983;"></button>`;
